.github/workflows/tests.yml

```
@@ -17,23 +17,37 @@ jobs:
     runs-on: ubuntu-latest
     container: returntocorp/ocaml:alpine
     steps:
-      - name: Install pipenv
-        run: sudo pip install pipenv
       - name: Pre-checkout fixes
         run: |
           sudo chmod -R 777 /github
           github_cache_dir="/__w"
           sudo mkdir -p "$github_cache_dir"
           sudo chmod -R 777 "$github_cache_dir"
-      - name: Make checkout speedy
-        run: git config --global fetch.parallel 50
       - name: Checkout
         uses: actions/checkout@v3
-        with:
-          submodules: true
+      - id: submodule-status
+        name: Get submodule version
+        run: echo ::set-output name=versions-hash::$(git submodule status --cached | sha256sum | cut -d" " -f1)
       - name: Post-checkout fixes
         run: ./.github/post-checkout
+      - name: Remove broken symlinks
+        run: find -L . -type l -exec rm {} \;
+      - id: core-cache
+        name: Check if this semgrep-core binary is cached
+        uses: actions/cache@v3
+        with:
+          path: ocaml-build-artifacts.tgz
+          key: "${{ runner.os }}-core-binary-${{ hashFiles('semgrep-core/**', '.github/workflows/tests.yml') }}-${{ steps.submodule-status.outputs.versions-hash }}"
```

should we add changes to root Makefile to the cache key?

---

```suggestion
          key: "${{ runner.os }}-core-binary-${{ hashFiles('semgrep-core/**', '.github/workflows/tests.yml', 'scripts/install-alpine-semgrep-core') }}-${{ steps.submodule-status.outputs.versions-hash }}"
```

Not the makefile cause it's not used for building here. But it's a good point that the install script should be part of this.